File: java/src/main/java/codingdojo/Session.java
package codingdojo;


import java.util.HashMap;
import java.util.Map;

/**
 * Allows the OnlineShopping to access data classes
 * and store them in the database
 */
public class Session {


    private final Map<String, ModelObject> session;

    public Session() {
        session = new HashMap<>();
        session.put("CART", new Cart());
        session.put("LOCATION_SERVICE", new LocationService());
    }

    public ModelObject get(String key) {
        return this.session.get(key);
    }

    public void put(String key, ModelObject value) {
        this.session.put(key, value);
    }
    
    public void saveAll() {
        for (String key : session.keySet()) {
            ModelObject entity = session.get(key);
            if (entity != null) {
                entity.saveToDatabase();
            }
        }
    }

    @Override
    public String toString() {
        StringBuffer sessionContents = new StringBuffer("\n");
        for (String key: session.keySet()) {
            sessionContents.append(key);
            sessionContents.append("=");
            ModelObject modelObject = session.get(key);
            sessionContents.append(modelObject);
            sessionContents.append("\n");
        }

        return "Session{" +
                sessionContents +
                "}";
    }
}

File: php/src/OnlineShoppingKata/Session.php
<?php

namespace App\OnlineShoppingKata;

/**
 * Class Session
 * @package OnlineShoppingKata
 *
 * Allows the OnlineShopping to access data classes
 * and store them in the database
 */
class Session
{
    private $session;

    /**
     * Session constructor.
     */
    public function __construct() {
        $this->session = [];
        $this->session["CART"] = new Cart();
        $this->session["LOCATION_SERVICE"] = new LocationService();
    }

    /**
     * @param string $key
     * @return ModelObject
     */
    public function get($key) {
        return $this->session[$key];
    }

    /**
     * @param $key
     * @param ModelObject $value
     */
    public function put($key, ModelObject $value) {
        $this->session[$key] = $value;
    }

    public function saveAll() {
        foreach (array_keys($this->session) as $array_key) {
            /** @var ModelObject $entity */
            $entity = $this->session[$array_key];
            $entity->saveToDatabase();
        }
    }

    /**
     * @return string
     */
    public function __toString() {
        $sessionContents = "\n";
        foreach ($this->session as $key => $value) {
            /** @var ModelObject $value */
            $sessionContents .= $key . "=" . $value . "\n";
        }

        return "Session{" . $sessionContents . "}";
    }
}